src/connection.c

```diff
@@ -240,18 +252,51 @@ int update_backoff_delay (backoff_timer_t *timer)
   return timer->delay;
 }  
 
-static void backoff_delay (backoff_timer_t *timer)
+#define BACKOFF_ERR -1
+#define BACKOFF_SHUTDOWN   1
+#define BACKOFF_DELAY_TAKEN 0
+
+void start_conn_in_progress (void);
+
+/* backoff_delay
+ * 
+ * delays for the number of seconds specified in parameter timer
+ * g_shutdown can break out of the delay.
+ *
+ * returns -1 pthread_cond_timedwait error
+ *  1   shutdown
+ *  0    delay taken
+*/  
+static int backoff_delay (backoff_timer_t *timer)
 {
-  update_backoff_delay (timer);
+  struct timespec ts;
+  int rtn;
 
-  // Update retry time for conn progress
-  if(timer->count == timer->max_count) 
-  {
-	start_conn_in_progress();
+  pthread_mutex_lock (&backoff_delay_mut);
+
+  clock_gettime (CLOCK_REALTIME, &ts);
+  if ((ts.tv_sec - timer->ts.tv_sec) >= 180) {
```

Comment:
As discussed, let's change 180s to 450s. Add comments that this is to update the retry time in health file at regular interval

Reply:
OK